src/detect-engine-event.c

```diff
@@ -342,44 +337,33 @@ static int EngineEventTestParse05 (void)
 static int EngineEventTestParse06 (void)
 {
     Packet *p = SCMalloc(SIZE_OF_PACKET);
-    if (unlikely(p == NULL))
-        return 0;
-    ThreadVars tv;
-    int ret = 0;
-    DetectEngineEventData *de = NULL;
-    SigMatch *sm = NULL;
+    FAIL_IF_NULL(p);
 
+    ThreadVars tv;
 
     memset(&tv, 0, sizeof(ThreadVars));
     memset(p, 0, SIZE_OF_PACKET);
 
     ENGINE_SET_EVENT(p,PPP_PKT_TOO_SMALL);
 
-    de = DetectEngineEventParse("decoder.ppp.pkt_too_small");
-    if (de == NULL)
-        goto error;
+    DetectEngineEventData *de = DetectEngineEventParse("decoder.ppp.pkt_too_small");
+    FAIL_IF_NULL(de);
 
     de->event = PPP_PKT_TOO_SMALL;
 
-    sm = SigMatchAlloc();
-    if (sm == NULL)
-        goto error;
+    SigMatch *sm = SigMatchAlloc();
+    FAIL_IF_NULL(sm);
 
     sm->type = DETECT_DECODE_EVENT;
     sm->ctx = (SigMatchCtx *)de;
 
-    ret = DetectEngineEventMatch(NULL,p,NULL,sm->ctx);
-
-    if(ret) {
-        SCFree(p);
-        return 1;
-    }
+    FAIL_IF_NOT(DetectEngineEventMatch(NULL, p, NULL, sm->ctx));
 
-error:
-    if (de) SCFree(de);
-    if (sm) SCFree(sm);
     SCFree(p);
-    return 0;
+    SCFree(de);
```

Comment:
Not sure if it makes a difference, but considering the call of commands in the original file, I'd go with:
```
SCFree(de);
SCFree(sm);
SCFree(p);

Reply:
I tries to follow the logical checks, then freeing the allocated pointers respectively.

I don't think the pointers are connected, or else there would've been an error running the unittests, 
or am I missing something?  
Should I make a change here?

Reply:
They may not be necessary, as your experience seems to indicate :) and indeed, except `sm` and `de`, the `packet` one doesn't look linked.

Let's leave it as is for now, and if there's anything wrong, it will be pointed out by Victor, before he approves... ;)